File: iosApp/iosApp/IosEchoAnalytics.swift
import Foundation
import FirebaseAnalytics
import FirebaseCrashlytics
import ComposeApp

class IosEchoAnalytics: EchoAnalytics {
    
    private let env: String
    
    init(env: String) {
        self.env = env
        Analytics.setUserProperty(env, forName: "env")
    }
    
    func logEvent(name: String, params: [String : Any]) {
        Analytics.logEvent(name, parameters: params)
    }
    
    func setUserId(id: String) {
        Analytics.setUserID(id)
        Crashlytics.crashlytics().setUserID(id)
    }
    
    func setUserProperty(name: String, value: String) {
        Analytics.setUserProperty(value, forName: name)
    }
    
    func logCrash(message: String) {
        Crashlytics.crashlytics().log(message)
    }
}

File: composeApp/src/iosMain/kotlin/com/bhanit/apps/echo/analytics/IosEchoAnalytics.kt
package com.bhanit.apps.echo.analytics

import com.bhanit.apps.echo.shared.analytics.EchoAnalytics

class IosEchoAnalytics : EchoAnalytics {
    override fun logEvent(name: String, params: Map<String, Any>) {
        println("IOS_ANALYTICS: Event '$name' with params: $params")
    }

    override fun setUserId(id: String) {
        println("IOS_ANALYTICS: setUserId: $id")
    }

    override fun setUserProperty(name: String, value: String) {
        println("IOS_ANALYTICS: setUserProperty: $name = $value")
    }

    override fun logCrash(message: String) {
        println("IOS_ANALYTICS: logCrash: $message")
    }
}
